helpers/query.go

```diff
@@ -0,0 +1,104 @@
+package helpers
+
+import (
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	tmclient "github.com/cosmos/cosmos-sdk/x/ibc/light-clients/07-tendermint/types"
+	"github.com/cosmos/relayer/relayer"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+// QueryBalance is a helper function for query balance
+func QueryBalance(chain *relayer.Chain, address string, showDenoms bool) (sdk.Coins, error) {
+	coins, err := chain.QueryBalanceWithAddress(address)
+	if err != nil {
+		return nil, err
+	}
+
+	if showDenoms {
+		return coins, nil
+	}
+
+	h, err := chain.QueryLatestHeight()
+	if err != nil {
+		return nil, err
+	}
+
+	dts, err := chain.QueryDenomTraces(0, 1000, h)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(dts.DenomTraces) > 0 {
+		out := sdk.Coins{}
+		for _, c := range coins {
+			for _, d := range dts.DenomTraces {
+				switch {
+				case c.Amount.Equal(sdk.NewInt(0)):
+				case c.Denom == d.IBCDenom():
+					out = append(out, sdk.NewCoin(d.GetFullDenomPath(), c.Amount))
+				default:
+					out = append(out, c)
+				}
+			}
+		}
+		return out, nil
+	}
+
+	return coins, nil
+}
+
+// QueryHeader is a helper function for query header
+func QueryHeader(chain *relayer.Chain, opts ...string) (*tmclient.Header, error) {
+	if len(opts) > 0 {
+		height, err := strconv.ParseInt(opts[0], 10, 64) //convert to int64
+		if err != nil {
+			return nil, err
+		}
+
+		if height == 0 {
```

Comment:
Ditto here, please check `if height <= 0` otherwise opts[0] == "-10" will slip through the cracks.